File: api/src/main/java/com/facundocastro/api/model/Proyecto.java
package com.facundocastro.api.model;

import com.sun.istack.NotNull;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
public class Proyecto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    private String titulo;
    private String descripcion;
    private String url;
    @Embedded
    private Imagen imagen;
}

File: angular-portfolio/src/app/models/Proyecto.ts
class Proyecto {
    protected _id: number = -1;
    protected _titulo: string = '';
    protected _descripcion: string = '';
    protected _url: string = '';
    protected _imagen: Imagen = new Imagen();

    public constructor() {}

    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    public get titulo(): string {
        return this._titulo;
    }
    public set titulo(value: string) {
        this._titulo = value;
    }

    public get descripcion(): string {
        return this._descripcion;
    }
    public set descripcion(value: string) {
        this._descripcion = value;
    }

    public get url(): string {
        return this._url;
    }
    public set url(value: string) {
        this._url = value;
    }

    public get imagen(): Imagen {
        return this._imagen;
    }
    public set imagen(value: Imagen) {
        this._imagen = value;
    }
}